src/client/mod.rs

```diff
@@ -255,20 +132,134 @@ impl TeeConnection {
+                    .unwrap();
+
+                if collateral_expired || !matches!(result, sgx_ql_qv_result_t::SGX_QL_QV_RESULT_OK)
+                {
+                    if collateral_expired {
+                        error!(
+                            "Collateral is out of date! Expired {}",
+                            earliest_expiration_date
+                        );
+                        return Err(Error::General(format!(
+                            "Collateral is out of date! Expired {}",
+                            earliest_expiration_date
+                        )));
+                    }
+
+                    let tcblevel = TcbLevel::from(result);
+                    if self
+                        .args
+                        .sgx_allowed_tcb_levels
+                        .map_or(true, |levels| !levels.contains(tcblevel))
+                    {
+                        error!("Quote verification result: {}", tcblevel);
+                        return Err(Error::General(format!(
+                            "Quote verification result: {}",
+                            tcblevel
+                        )));
+                    }
+
+                    info!("TcbLevel is allowed: {}", tcblevel);
+                }
+
+                for advisory in advisories {
+                    warn!("Info: Advisory ID: {advisory}");
+                }
+
+                if let Some(mrsigner) = &self.args.sgx_mrsigner {
+                    let mrsigner_bytes = hex::decode(mrsigner)
+                        .map_err(|e| Error::General(format!("Failed to decode mrsigner: {}", e)))?;
+                    if quote.report_body.mrsigner[..] != mrsigner_bytes {
+                        return Err(Error::General(format!(
+                            "mrsigner mismatch: got {}, expected {}",
+                            hex::encode(quote.report_body.mrsigner),
+                            &mrsigner
+                        )));
+                    } else {
+                        info!("mrsigner `{mrsigner}` matches");
+                    }
+                }
+
+                if let Some(mrenclave) = &self.args.sgx_mrenclave {
+                    let mrenclave_bytes = hex::decode(mrenclave).map_err(|e| {
+                        Error::General(format!("Failed to decode mrenclave: {}", e))
+                    })?;
+                    if quote.report_body.mrenclave[..] != mrenclave_bytes {
+                        return Err(Error::General(format!(
+                            "mrenclave mismatch: got {}, expected {}",
+                            hex::encode(quote.report_body.mrenclave),
+                            &mrenclave
+                        )));
+                    } else {
+                        info!("mrenclave `{mrenclave}` matches");
+                    }
+                }
+
+                Ok(rustls::client::danger::ServerCertVerified::assertion())
```

Comment:
Is this just a weird API? Calling `assertion()` here feels wrong. Looking at the API, I guess they want to make it stick out.

Reply:
yeah, it is weird